boa_engine/src/object/jsmap.rs

```diff
@@ -0,0 +1,344 @@
+//! This module implements a wrapper for the Map Builtin Javascript Object
+use crate::{
+    builtins::map::{add_entries_from_iterable, ordered_map::OrderedMap},
+    builtins::Map,
+    object::{JsFunction, JsMapIterator, JsObject, JsObjectType, ObjectData},
+    Context, JsResult, JsValue,
+};
+
+use boa_gc::{Finalize, Trace};
+use std::ops::Deref;
+
+/// `JsMap` provides a wrapper for Boa's implementation of the Javascript `Map` object.
+///
+/// # Examples
+///
+/// Create a `JsMap` and set a new entry
+/// ```
+/// fn main() -> JsResult<()> {
+///     // Create default `Context`
+///     let context = &mut Context::default();
+///     // Create a new empty `JsMap`.
+///     let map = JsMap::new(context);
+///
+///     // Set key-value pairs for the `JsMap`.
+///     map.set("Key-1", "Value-1", context)?;
+///     map.set("Key-2", 10, context)?;
+///
+///     assert_eq!(map.get_size(context)?, 2.into());
+///
+///     Ok(())
+/// }
```

Comment:
You can remove the `main` definition and use `unwrap` instead of `?`, since the functionality is the emphasis of the code snippet and not the error handling.
And just to clarify: doc tests automatically wrap the snippet in a `main` function, meaning you can omit it on all the doc examples

Reply:
Also, if you do want to preserve the error handling, the rustdoc book has an explanation on how to do that:

https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#using--in-doc-tests

It's a bit more cumbersome but it would probably look better than using tons of `unwrap`s

Reply:
Thanks for the links to the docs! I swear I was looking for those, but I think I was in a completely different book. I believe I managed to clean up all the examples and add all the `unwrap`s where they were needed.